### DISCUSSION TITLE
Confusion between icub-firmware and icub-firmware-build

### DISCUSSION THREAD

**Discussion by pedrombmachado** [robotology/community#195] [Q&A] [answered]
Hi all,

I am trying to fix my environmental variables and I have a question. On the [wiki](http://wiki.icub.org/wiki/ICub_PC104_Linux_Image_-_Detailed_info) it is said:

```sh
export FIRMWARE_BIN=${ROBOT_CODE}/icub-firmware/build
```

However, the repository name is [icub-firmware-build](https://github.com/robotology/icub-firmware-build) and therefore shouldn't it be:

```sh
export FIRMWARE_BIN=${ROBOT_CODE}/icub-firmware-build
```

Many thanks,
Pedro

**Comment by pedrombmachado** [comment]
Found the answer. I was missing the icub-firmware repository.

**Comment by pedrombmachado** [comment]
Found the [repository](https://github.com/robotology/icub-firmware) however it does not contain the build folder. 

Shall I leave it like this?

```sh
export FIRMWARE_BIN=${ROBOT_CODE}/icub-firmware
```

**Comment by randaz81** [comment, marked as answer]
- `icub-firmware` contains the sources of the boards firmware. You don't need to download that repo, it is simply useless unless you have the firmware compiler(s).
- `icub-firmware-build` contains the binaries, generated from `icub-firmware` sources. It's the folder you need to check when you want to upgrade the firmware of robot boards.
 - `icub-firmware-shared` contains some sources and definitions needed to compile some plugins of `icub-main` necessary to run `yarprobotinterface` on the PC104 (e.g. `embObjMotionControl`).

Currently, I'm not aware of any practical use of the environment variable `FIRMWARE_BIN`. I think it has been reserved by developers for future use, so please simply ignore it.

**Comment by pedrombmachado** [comment]
Hi @randaz81 
  Many thanks for the reply. If possible please update the [Readme](https://github.com/robotology/icub-firmware-shared) to reflect that.
Many thanks,
Pedro

**Comment by traversaro** [comment]
I opened a PR for adding a README to `icub-firmware-build` one year ago explicitly to avoid this confusion: https://github.com/robotology/icub-firmware-build/pull/8 , nobody ever commented, so I think we can merge it. : )

**Comment by pattacini** [comment]
Go ahead 😉 then @traversaro
